tensordict/_torch_func.py

```diff
@@ -929,15 +929,11 @@ def _grad(
         )
 
     if grad_outputs is not None:
-        tup_grad_outputs = tuple(
-            grad_outputs._values_list(True, True, is_leaf=_NESTED_TENSORS_AS_LISTS)
-        )
+        tup_grad_outputs = tuple(grad_outputs[k] for k in outputs.keys(True, True))
```

Comment:
When using a lazy stack, `grad_outputs._values_list(..., is_leaf=_NESTED_TENSORS_AS_LISTS)` will return the leaf tensors (not their stack).
It's important if they can't be stacked (ragged dimensions).
The new implementation will break this I think (try with a stack of tensordicts that have different tensor shapes to convince yourself!)